### PR TITLE
charge_point: Reject ChargingStationExternalConstraints profiles in SetChargingProfileRequest

### PR DESCRIPTION
**Pull request by christopher-davis-afs**
## Describe your changes

Implements K01.FR.22

## Issue ticket number and link

https://github.com/US-JOET/base-camp/issues/164

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] If OCPP 2.0.1: I have updated the [OCPP 2.0.1 status document](https://github.com/EVerest/libocpp/tree/main/doc/ocpp_201_status.md)
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

File: lib/ocpp/v201/charge_point.cpp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // SPDX-License-Identifier: Apache-2.0
 // Copyright 2020 - 2023 Pionix GmbH and Contributors to EVerest
 
+#include "ocpp/v201/ocpp_types.hpp"
```
Comment (ghost) [inline lib/ocpp/v201/charge_point.cpp]: NIT: I think this is in the header so could be removed here.

File: lib/ocpp/v201/charge_point.cpp [hunk 2]
```diff
@@ -3161,6 +3162,19 @@ void ChargePoint::handle_set_charging_profile_req(Call<SetChargingProfileRequest
     SetChargingProfileResponse response;
     response.status = ChargingProfileStatusEnum::Rejected;
 
+    if (msg.chargingProfile.chargingProfilePurpose == ChargingProfilePurposeEnum::ChargingStationExternalConstraints) {
+        response.statusInfo = StatusInfo();
+        response.statusInfo->reasonCode = "InvalidValue";
```
Comment (ghost) [inline lib/ocpp/v201/charge_point.cpp:3166]: NIT: Could the reasonCode be InvalidProfileType? Not sure how we keep this information consistent.

**Review by ghost — APPROVED** [review]
LGTM - ☕

**Review by gberardi-pillar — APPROVED** [review]

File: lib/ocpp/v201/charge_point.cpp (hunk 2 above)
Comment (christopher-davis-afs) [reply lib/ocpp/v201/charge_point.cpp:3166]: That isn't one of the accepted reason codes listed in the OCPP appendices

**Pull request merged**